autolab/Autolab
# Fix Yaml file in assessment not being updated properly after import, ensure proper loading of version penalty and late penalty

## Summary

### Summary generated by Reviewpad on 27 Jan 24 22:52 UTC
This pull request includes two patches. 

The first patch fixes an issue where the YAML fields for the scoreboard and autograder are not properly dumped during import. It also addresses the problem with the late penalty and version penalty not being loaded. The changes include updating the method `deserialize` in the `Assessment` model to properly handle the loading and updating of these fields.

The second patch adds a test case to ensure that the dumped YAML file after import matches the exported file. It also updates the `rails_helper` file to set the `file_fixture_path` to match the `fixture_path`, which is necessary for Rails 7.0.


## Description

- Assign autograder, scoreboard, version penalty, late penalty to assessment in `deserialize` so that they accurately get dumped by `dump_yaml` / don't get overwritten by `invalidate_raw_scores`
- Add test that verifies that yaml in the imported assessment matches the tar file it was imported from

There is some weird behavior going on with `invalidate_raw_scores`, but I could fully understand the nature of how it was causing the yaml file to be overwritten without the autograder and scoreboard fields. However, in `deserialize`, because the above fields weren't being assigned to the assessment, they weren't really being properly imported regardless, so this should fix things. 

## Motivation and Context


Closes #2053

## How Has This Been Tested?



- Export an assessment with a scoreboard, autograder, late penalty, version penalty, etc (here is one [hellotar-2_test.tar.zip](https://github.com/autolab/Autolab/files/14074228/hellotar-2_test.tar.zip))
- See that these fields are included in the export like so:
```
---
autograder:
  autograde_image: autograding_image
  autograde_timeout: 180
  release_score: true
dates:
  due_at: '2023-05-18 12:05:00 -0400'
  end_at: '2023-05-18 12:05:00 -0400'
  start_at: '2023-05-18 12:05:00 -0400'
general:
  allow_student_assign_group: true
  category_name: test
...
  max_size: 2
  max_submissions: -1
  version_threshold: 4
  writeup: ''
late_penalty:
  kind: 0
  value: 2.0
problems:
- description: ''
  max_score: 0.0
  name: Problem 1
  optional: false
...
scoreboard:
  banner: bruh moment!
  colspec: ''
version_penalty:
  kind: 1
  value: 5.0
 ```
- Import the tar into a course
- View the imported assessment yaml file by going to (using file explorer, terminal etc) `courses/{course}/assessment/assessment.yml` inside Autolab repo
- See that yaml includes the scoreboard, autograder, version_penalty, late_penalty fields (also everything else)

There is also an automated test that verifies this above workflow included in the PR

## Types of changes


## Checklist:


## Other issues / help required




If unsure, feel free to submit first and we'll help you along.
